### PR TITLE
add tests for validating ragged input support for ort

### PR DESCRIPTION
**Pull request by askhade**

### CODE REVIEW

**Review by Sifat1998 — DISMISSED** [review]

File: qa/common/gen_qa_ragged_models.py [hunk 1]
```diff
@@ -254,6 +279,79 @@ def create_plan_modelfile(models_dir, model_version, dtype):
...
+    in0 = onnx.helper.make_tensor_value_info("RAGGED_INPUT", onnx_dtype,
+                                             tu.shape_to_onnx_shape([-1]))
+
+    bs_in = onnx.helper.make_tensor_value_info("BATCH_AND_SIZE_INPUT", onnx_dtype,
+                                             tu.shape_to_onnx_shape([-1]))
+
+    batch_in = onnx.helper.make_tensor_value_info("BATCH_INPUT", onnx_dtype,
+                                             tu.shape_to_onnx_shape([-1]))
+
+    out = onnx.helper.make_tensor_value_info("RAGGED_OUTPUT", onnx_dtype, tu.shape_to_onnx_shape([-1, -1]))
```
Comment (askhade) [inline qa/common/gen_qa_ragged_models.py:312]: @GuanLuo : Can you verify the output shapes

File: qa/common/gen_qa_ragged_models.py [hunk 2]
```diff
@@ -351,6 +449,10 @@ def create_batch_input_models(models_dir):
         create_modelconfig(models_dir, 4, model_version, np.float32,
                            "tensorflow", "savedmodel")
         create_savedmodel_modelfile(models_dir, model_version, np.float32)
+    if FLAGS.onnx:
+        create_modelconfig(models_dir, 4, model_version, np.float32,
+                           "onnx", "onnx")
```
Comment (askhade) [inline qa/common/gen_qa_ragged_models.py:454]: @GuanLuo : can you confirm for ort backend and platform name is onnx

Comment (askhade) [reply qa/common/gen_qa_ragged_models.py:454]: Also can you please help me understand if the config needs any onnx specific changes? and also point me to some documentation where I can understand these requirements. Thanks!

**Comment by askhade** [comment]
marked this PR as WIP because there are some minor updates which are still remaining on my side and also because I am working on verifying whether these test run with the changes in onnxruntime backend.  Wanted to send this PR so that you folks can take a look and let me know if something obvious is missing.

**Review by deadeyegoodwin — DISMISSED** [review]

File: qa/common/gen_qa_ragged_models.py (hunk 2 above)
Comment (GuanLuo) [reply qa/common/gen_qa_ragged_models.py]: backend should be "onnxruntime", and the platform can be left empty as ORT backend doesn't require it for further distinguishment. I don't think ONNX requires specific changes.

File: qa/common/gen_qa_ragged_models.py [hunk 3]
```diff
@@ -254,6 +280,108 @@ def create_plan_modelfile(models_dir, model_version, dtype):
...
+    # Because the ragged input and batch input are one dimensional vector
+    # when passing to the model, the model must generate output with batch
+    # dimension so that Triton can scatter it to different responses along
+    # the batch dimension.
+    # 'BATCH_AND_SIZE_INPUT' is also used as a hint to generate output with
+    # batch dimension, 'BATCH_AND_SIZE_INPUT' must have shape [batch_size].
+    # Each output corresponds to the input with the same name, so if there
+    # are two requests, one has "RAGGED_INPUT" [2, 4] and the other has [1],
+    # since the input is ragged, the model sees the input as [2, 4, 1], and
+    # "BATCH_AND_SIZE_INPUT" will have shape [3]. Then the model output will
```
Comment (GuanLuo) [inline qa/common/gen_qa_ragged_models.py:294]: `"BATCH_AND_SIZE_INPUT" will have shape [2]`, I think this typo also appears in other `create_xx_modelfile()`, do you mind to update those as well?

**Review by GuanLuo — DISMISSED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**